src/components/LinkButton/LinkButton.tsx

```diff
@@ -0,0 +1,60 @@
+import React, { ComponentProps, DetailedHTMLProps } from 'react';
+import { Link, LinkProps } from 'react-router-dom';
+import { assert } from 'common/utils';
+import { StyledButton } from './LinkButton.style';
+import { EventAction, EventCategory, trackEvent } from 'components/Analytics';
+
+type AnchorLinkType = DetailedHTMLProps<
+  React.AnchorHTMLAttributes<HTMLAnchorElement>,
+  HTMLAnchorElement
+>;
+
+type StyledButtonProps = ComponentProps<typeof StyledButton>;
+
+type ExternalLinkButtonProps = StyledButtonProps & AnchorLinkType;
+
+type InternalLinkButtonProps = StyledButtonProps & LinkProps;
+
+interface TrackingProps {
+  trackingCategory: EventCategory;
+  trackingAction?: EventAction;
+  trackingLabel: string;
+}
+
+export type LinkButtonProps = InternalLinkButtonProps | ExternalLinkButtonProps;
+
+const LinkButton: React.FC<LinkButtonProps & TrackingProps> = props => {
+  assert(
+    props.href || props.to,
+    "LinkButton needs either 'to' or 'href' as props",
+  );
+
+  const isInternalLink = props.to;
+  const { trackingAction, trackingCategory, trackingLabel } = props;
+
+  const onClick = (ev: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
+    const action =
+      trackingAction || isInternalLink
+        ? EventAction.NAVIGATE
+        : EventAction.CLICK_LINK;
```

Comment:
Originally, I wanted to have `CLICK` to be for clicks on buttons and other UI elements, and `CLICK_LINK` for external links only. I didn't think that `NAVIGATE` would be needed because we can always see page views (and sequences of page views) on GA, but I think that having a `NAVIGATE` event might make reporting easier in some cases.

Since the LinkButton is always intended for internal or external navigation, `NAVIGATE` and `CLICK_LINK` should be enough, but in some cases we use a different action type.

One example is the vaccination banner, where we do use `CLICK` for internal navigation, which makes sense because clicking the buttons does change the UI, thats's why trackingAction is an optional prop.